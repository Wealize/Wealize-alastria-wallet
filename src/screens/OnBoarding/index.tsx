import React from 'react'
import { useWindowDimensions } from 'react-native'
import { NavigationFunctionComponent } from 'react-native-navigation'

import WhiteButton from '../../components/WhiteButton'
import { Colors } from '../../utils/themes'
import { NavigationProps, setStackRoot } from '../../utils/navigation-utils'
import {
  BottomContainer,
  Subtitle,
  Title,
  ImageScreen,
  IconHeader,
  LinkHeader,
  TextHeader,
  SubtitleAlt,
  SwiperStyled,
  MainView,
  TitleContainer,
  SkipButton,
  SkipButtonAlt
} from './styles'
import { ONBOARDING } from '../../constants/text'
import { IMG } from '../../constants/urlImages'
import { SCREEN } from '../../constants/screens'

const OnBoarding: NavigationFunctionComponent = ({
  componentId
}: NavigationProps) => {
  return (
    <MainView>
      <SwiperStyled
        width={useWindowDimensions().width}
        height={useWindowDimensions().height}
        loop={false}
      >
        <ImageScreen source={IMG.ONBOARDING_BACKGROUND.FIRST_SCREEN} style={{}}>
          <LinkHeader>
            <IconHeader source={IMG.ONBOARDING_BACKGROUND.LINK_HEADER} />
            <TextHeader>{ONBOARDING.LINK_HEADER}</TextHeader>
          </LinkHeader>
          <TitleContainer>
            <Title>{ONBOARDING.FIRST_SCREEN.TITLE}</Title>
            <SkipButton
              onPress={() => setStackRoot(componentId, SCREEN.REGISTER)}
            >
              {ONBOARDING.SKIP}
            </SkipButton>
          </TitleContainer>
          <Subtitle>{ONBOARDING.FIRST_SCREEN.SUBTITLE}</Subtitle>
        </ImageScreen>

        <ImageScreen
          source={IMG.ONBOARDING_BACKGROUND.SECOND_SCREEN}
          style={{}}
        >
          <LinkHeader>
            <IconHeader source={IMG.ONBOARDING_BACKGROUND.LINK_HEADER} />
            <TextHeader>{ONBOARDING.LINK_HEADER}</TextHeader>
          </LinkHeader>
          <TitleContainer>
            <Title>{ONBOARDING.SECOND_SCREEN.TITLE}</Title>
            <SkipButtonAlt
              onPress={() => setStackRoot(componentId, SCREEN.REGISTER)}
            >
              {ONBOARDING.SKIP}
            </SkipButtonAlt>
          </TitleContainer>
          <SubtitleAlt>{ONBOARDING.SECOND_SCREEN.SUBTITLE}</SubtitleAlt>
        </ImageScreen>

        <ImageScreen source={IMG.ONBOARDING_BACKGROUND.THIRD_SCREEN} style={{}}>
          <LinkHeader>
            <IconHeader source={IMG.ONBOARDING_BACKGROUND.LINK_HEADER} />
            <TextHeader>{ONBOARDING.LINK_HEADER}</TextHeader>
          </LinkHeader>
          <Title>{ONBOARDING.THIRD_SCREEN.TITLE}</Title>
          <Subtitle>{ONBOARDING.THIRD_SCREEN.SUBTITLE}</Subtitle>
          <BottomContainer>
            <WhiteButton
              title={ONBOARDING.BUTTON}
              titleStyle={{ color: Colors.principalButtons }}
              onPress={() => setStackRoot(componentId, SCREEN.REGISTER)}
            />
          </BottomContainer>
        </ImageScreen>
      </SwiperStyled>
    </MainView>
  )
}

OnBoarding.options = {
  statusBar: {
    backgroundColor: Colors.mainBackground,
    style: 'dark'
  },
  topBar: {
    visible: false
  }
}

export default OnBoarding
