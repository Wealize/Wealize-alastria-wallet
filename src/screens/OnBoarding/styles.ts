import styled from 'styled-components'
import { View, Text, ImageBackground, Image } from 'react-native'
import Swiper from 'react-native-swiper'

import { SUBTITLE_FONT, TITLE_FONT } from '../../styles/FontFamilies'
import { Colors } from '../../utils/themes'

export const MainView = styled(View)`
  flex: 1;
  flex-direction: column;
`

export const SwiperStyled = styled(Swiper).attrs({
  activeDotColor: Colors.backgroundButtons,
  paginationStyle: { bottom: '15%' }
})``

export const ImageScreen = styled(ImageBackground).attrs({
  imageStyle: { resizeMode: 'cover' }
})`
  width: 100%;
  height: 100%;
`

export const LinkHeader = styled(Text)`
  margin-top: 10%;
  margin-left: 7.5%;
  margin-bottom: 3%;
`

export const TextHeader = styled(Text)`
  margin-top: 10%;
  margin-left: 8%;
  font-family: ${TITLE_FONT};
  font-weight: 600;
  font-size: 14px;
  color: ${Colors.normalText};
`

export const IconHeader = styled(Image)`
  width: 24px;
  height: 24px;
`

export const TitleContainer = styled(View)`
  flex-direction: row;
  justify-content: space-between;
`

export const Title = styled(Text)`
  width: 50%;
  margin-left: 9%;
  margin-bottom: 7%;
  font-family: ${TITLE_FONT};
  font-weight: bold;
  font-size: 20px;
  color: ${Colors.normalText};
`

export const SkipButton = styled(Text)`
  font-size: 13px;
  margin-right: 9%;
  color: ${Colors.textButton};
`

export const SkipButtonAlt = styled(SkipButton)`
  color: ${Colors.normalText};
`

export const Subtitle = styled(Text)`
  width: 45%;
  margin-left: 9%;
  margin-top: 20px;
  font-family: ${SUBTITLE_FONT};
  font-size: 14px;
  color: ${Colors.normalText};
`

export const SubtitleAlt = styled(Subtitle)`
  margin-top: 0px;
`

export const BottomContainer = styled(View)`
  position: absolute;
  left: 25%;
  bottom: 8%;
`
