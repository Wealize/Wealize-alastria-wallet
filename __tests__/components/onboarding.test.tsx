import React from 'react'
import renderer from 'react-test-renderer'

import OnBoarding from '../../src/screens/OnBoarding'


describe('OnboardginComponent', () => {
  it('should render correctly', () => {
    const result = renderer.create(<OnBoarding componentId={'1'} />)
    expect(result).toMatchSnapshot()
  })
})