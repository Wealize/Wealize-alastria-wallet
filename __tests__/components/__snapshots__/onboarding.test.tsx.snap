// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardginComponent should render correctly 1`] = `
<View
  className="sc-gsTEea lcyMyB"
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      activeDotColor="#ffffff"
      automaticallyAdjustContentInsets={false}
      autoplay={false}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      className="sc-dlfnuX"
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      contentOffset={
        Object {
          "x": 0,
        }
      }
      disableNextButton={false}
      disablePrevButton={false}
      height={1334}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={false}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      paginationStyle={
        Object {
          "bottom": "15%",
        }
      }
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
      width={750}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={Object {}}
          >
            <Image
              className="sc-hKgJUU kXhtQk"
              source={
                Object {
                  "testUri": "../../../src/assets/img/onboarding-first-screen.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "resizeMode": "cover",
                  },
                ]
              }
            />
            <Text
              className="sc-eCstlR cJeqmM"
            >
              <Image
                className="sc-gKseQn jVgprL"
                source={
                  Object {
                    "testUri": "../../../src/assets/img/icon-info.png",
                  }
                }
              />
              <Text
                className="sc-jSgvzq iCgIlM"
              >
                Más información
              </Text>
            </Text>
            <View
              className="sc-iBPTik hkBWhy"
            >
              <Text
                className="sc-fubCzh hBDdKR"
              >
                Identfy
              </Text>
              <Text
                className="sc-pGacB ezXVdE"
                onPress={[Function]}
              >
                Saltar
              </Text>
            </View>
            <Text
              className="sc-kEjbQP gTxvZb"
            >
              La aplicación que te permite guardar de forma segura tu documentación en el móvil.
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={Object {}}
          >
            <Image
              className="sc-hKgJUU kXhtQk"
              source={
                Object {
                  "testUri": "../../../src/assets/img/onboarding-second-screen.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "resizeMode": "cover",
                  },
                ]
              }
            />
            <Text
              className="sc-eCstlR cJeqmM"
            >
              <Image
                className="sc-gKseQn jVgprL"
                source={
                  Object {
                    "testUri": "../../../src/assets/img/icon-info.png",
                  }
                }
              />
              <Text
                className="sc-jSgvzq iCgIlM"
              >
                Más información
              </Text>
            </Text>
            <View
              className="sc-iBPTik hkBWhy"
            >
              <Text
                className="sc-fubCzh hBDdKR"
              >
                Guarda o comparte información
              </Text>
              <Text
                className="sc-pGacB sc-jrAFXE ezXVdE ksFMZZ"
                onPress={[Function]}
              >
                Saltar
              </Text>
            </View>
            <Text
              className="sc-kEjbQP sc-iqHYmW gTxvZb iQNRH"
            >
              Utiliza el escáner de la aplicación para escanear un código QR y guardar la información acreditada que te emitan las organizaciones. Usa el mismo procedimiento cuando una organización te solicite información, y podrás decidir qué información deseas compartir.
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 1334,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={Object {}}
          >
            <Image
              className="sc-hKgJUU kXhtQk"
              source={
                Object {
                  "testUri": "../../../src/assets/img/onboarding-third-screen.png",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  Object {
                    "resizeMode": "cover",
                  },
                ]
              }
            />
            <Text
              className="sc-eCstlR cJeqmM"
            >
              <Image
                className="sc-gKseQn jVgprL"
                source={
                  Object {
                    "testUri": "../../../src/assets/img/icon-info.png",
                  }
                }
              />
              <Text
                className="sc-jSgvzq iCgIlM"
              >
                Más información
              </Text>
            </Text>
            <Text
              className="sc-fubCzh hBDdKR"
            >
              Revocar información
            </Text>
            <Text
              className="sc-kEjbQP gTxvZb"
            >
              En el momento que desees dejar de compartir información con una institución, puedes revocar su acceso desde el listado de accesos. Accede a él desde el menú de navegación inferior.
            </Text>
            <View
              className="sc-crrszt Chhjb"
            >
              <View
                style={
                  Array [
                    Object {
                      "borderRadius": 3,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 3,
                    },
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "busy": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderColor": "#ca2b63",
                        "borderRadius": 45,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "height": 39,
                        "justifyContent": "center",
                        "padding": 8,
                        "width": 224,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#ca2b63",
                          "fontSize": 18,
                          "paddingVertical": 1,
                          "textAlign": "center",
                        }
                      }
                    >
                      Comenzar
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 25,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "bottom": "15%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,.2)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,.2)",
              "borderRadius": 4,
              "height": 8,
              "marginBottom": 3,
              "marginLeft": 3,
              "marginRight": 3,
              "marginTop": 3,
              "width": 8,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
